### PR TITLE
wtype: 2020-09-14 -> 0.3

### PR DESCRIPTION
**Pull request by roosemberth**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/wayland/wtype/default.nix [hunk 1]
```diff
@@ -10,18 +11,18 @@
...
-  version = "2020-09-14";
+  version = "0.3";
 
   src = fetchFromGitHub {
     owner = "atx";
     repo = "wtype";
-    rev = "74071228dea4047157ae82960a2541ecc431e4a1";
-    sha256 = "1ncspxpnbwv1vkfmxs58q7aykjb6skaa1pg5sw5h798pss5j80rd";
+    rev = "v${version}";
+    sha256 = "sha256-8q2PxI3ItR4tsP/KOiSlqnuK4ZSe7OdekheolEFVmys=";
```
Comment (worldofpeace) [inline pkgs/tools/wayland/wtype/default.nix:22]:
```suggestion
    hash = "sha256-8q2PxI3ItR4tsP/KOiSlqnuK4ZSe7OdekheolEFVmys=";
```

Comment (roosemberth) [reply pkgs/tools/wayland/wtype/default.nix]: Whoops, sorry.

Thanks!

Comment (SuperSandro2000) [reply pkgs/tools/wayland/wtype/default.nix]: > Whoops, sorry.
> 
> Thanks! (fixed)

It does not matter if it attribute is named sha256 or hash in this case but hash is nicer.

**Review by justinlovinger — APPROVED** [review]
Looks good to me.

File: pkgs/tools/wayland/wtype/default.nix [hunk 2]
```diff
@@ -10,18 +11,18 @@
...
     owner = "atx";
     repo = "wtype";
-    rev = "74071228dea4047157ae82960a2541ecc431e4a1";
-    sha256 = "1ncspxpnbwv1vkfmxs58q7aykjb6skaa1pg5sw5h798pss5j80rd";
+    rev = "v${version}";
+    hash = "sha256-8q2PxI3ItR4tsP/KOiSlqnuK4ZSe7OdekheolEFVmys=";
   };
 
-  nativeBuildInputs = [ meson ninja pkg-config wayland ];
+  nativeBuildInputs = [ meson ninja pkg-config wayland git ];
```
Comment (SuperSandro2000) [inline pkgs/tools/wayland/wtype/default.nix:25]: Why is it requiring git? We do not have any git directory in the source.

Comment (JustinLovinger) [reply pkgs/tools/wayland/wtype/default.nix:25]: It looks like the wtype meson build uses `git`: <https://github.com/atx/wtype/blob/master/meson.build#L11>.

Comment (roosemberth) [reply pkgs/tools/wayland/wtype/default.nix]: I didn't realize it was an optional dependency. I vaguely recall nix-build failed without it, but after removing nix-build is still happy and the app works (for my limited test of just a couple of keystrokes).

I've updated the merge request accordingly.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>rofimoji</li>
<li>wtype</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

rofimoji:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Pull request merged**